vllm/multimodal/inputs.py

```diff
@@ -235,8 +238,15 @@ def __eq__(self, other: object) -> bool:
         if not isinstance(other, self.__class__):
             return False
 
+        if self.data is None:
+            data_equal = other.data is None
+        elif other.data is None:
+            data_equal = self.data is None
```

Comment:
```suggestion
        if self.data is None or other.data is None:
            data_equal = self.data == other.data
```
I think we can simplify the condition here.

Reply:
If I change to this, type checking fails because the linter thinks that `data_equal` can be a tensor